src/System.Private.CoreLib/shared/System/Reflection/LocalVariableInfo.cs

```diff
@@ -0,0 +1,25 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+
+namespace System.Reflection
+{
+    public class LocalVariableInfo
+    {        
+        protected LocalVariableInfo() { }
+        public virtual Type LocalType { get { Debug.Fail("type must be set!"); return null; } }
+        public virtual bool IsPinned => false;
+        public virtual int LocalIndex => 0;
+
+        public override string ToString()
+        {
+            // This is really how the desktop behaves if you don't override, including the NullReference when 
+            // it calls ToString() on LocalType's null return.
+            string toString = LocalType.ToString() + " (" + LocalIndex + ")";
+            return IsPinned ? toString += " (pinned)" : toString;
```

Comment:
I think that the CoreCLR implementation of this is easier to understand.

Reply:
I agree. IMO there are way too many operations taking place in this one line.

Reply:
@jkotas Actually does 'pinned' need to be localized? I remember you mentioning that words like 'blittable' are localized, I would assume 'pinned' is also localizable.

Reply:
Right. It has to match https://github.com/dotnet/coreclr/blob/fcb04373e2015ae12b55f33fdd0dd4580110db98/src/vm/runtimehandles.h#L102

Reply:
BTW: it should be fine to change isPinned field to bool (on both managed and unmamaged sides) to make this look better.

Reply:
The localization for these words is not consistent. Since it was not localized in desktop here, I would keep it as is.